### PR TITLE
Caffe rc3 built using the foss 2016a toolchain.

### PR DESCRIPTION
**Pull request by FokkeDijkstra**
Caffe rc3 built using the foss 2016a toolchain.

### CODE REVIEW

**Comment by FokkeDijkstra** [comment]
Should I include snappy using the foss-2016a toolchain or go for --minimal-toolchains?
If I do the latter I get a complaint about protobuf-python, which I don't understand:

`ERROR: Failed to process easyconfig /home/fokke/easybuild/easybuild-easyconfigs/easybuild/easyconfigs/c/Caffe/Caffe-rc3-foss-2016a-CUDA-7.5.18-Python-2.7.11.eb: No easyconfig for {'external_module_metadata': {}, 'full_mod_name': None, 'build_only': False, 'versionsuffix': '-Python-%(pyver)s', 'external_module': False, 'dummy': False, 'toolchain': None, 'name': 'protobuf-python', 'short_mod_name': None, 'version': '3.0.2', 'hidden': False} that matches toolchain hierarchy generated by {'version': '2016a', 'name': 'foss'}`

**Comment by FokkeDijkstra** [comment]
I am still wondering of minimal toolchains should have fixed this issue. Is this active now? Can the test be run again if this is the case?

File: easybuild/easyconfigs/c/Caffe/Caffe-rc3-foss-2016a-CUDA-7.5.18-Python-2.7.11.eb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+homepage = 'https://github.com/BVLC/caffe'
+description = """
+Caffe is a deep learning framework made with expression, speed, 
+and modularity in mind. It is developed by the Berkeley Vision 
+and Learning Center (BVLC) and community contributors.
+"""
+
+toolchain = {'name': 'foss', 'version': '2016a'}
+toolchainopts = {'optarch': False, 'pic': True}
```
Comment (boegel) [inline easybuild/easyconfigs/c/Caffe/Caffe-rc3-foss-2016a-CUDA-7.5.18-Python-2.7.11.eb]: @FokkeDijkstra any reason `optarch` is being hard disabled here?

**Comment by boegel** [comment]
@FokkeDijkstra Since EasyBuild v3.0, `--robot` will also consider subtoolchains when resolving dependencies, so you can contribute easyconfigs that worked fine under `--minimal-toolchains`

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (1 easyconfigs in this PR)
node2605.swalot.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/86d0bb1de355d3b2635f03b8cd172732 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2086.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/38992e7239112b2d5e46b145eb28af0f for a full test report.

File: easybuild/easyconfigs/c/Caffe/Caffe-rc3-foss-2016a-CUDA-7.5.18-Python-2.7.11.eb (hunk 1 above)
Comment (FokkeDijkstra) [inline easybuild/easyconfigs/c/Caffe/Caffe-rc3-foss-2016a-CUDA-7.5.18-Python-2.7.11.eb]: I think it slipped in while debugging the build process. I've removed it, and the software still builds fine.

Comment (boegel) [inline easybuild/easyconfigs/c/Caffe/Caffe-rc3-foss-2016a-CUDA-7.5.18-Python-2.7.11.eb]: @FokkeDijkstra please update the pull request to only enable `pic`?

```python
toolchainopts = {'pic': True}
```

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2111.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/62d32e84d29b53f2805d4fe1919ab6f9 for a full test report.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2625.swalot.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/5e120071801dbc722d55750da6e31931 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2625.swalot.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/b1f72d485f988a4df60f2450aaa0e729 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @FokkeDijkstra!

**Pull request merged**